tests/pyfunc/docker/test_docker_flavors.py

```diff
@@ -109,14 +109,16 @@ def stream_logs():
                 response = requests.get(url=f"http://localhost:{port}/ping")
                 if response.ok:
                     break
-            except requests.exceptions.ConnectionError:
+            except requests.exceptions.ConnectionError as e:
+                sys.stdout.write(f"An exception occurred when calling the server: {e}\n")
                 time.sleep(5)
 
             container.reload()  # update container status
             if container.status == "exited":
                 raise Exception("Container exited unexpectedly.")
 
             sys.stdout.write(f"Container status: {container.status}\n")
+            time.sleep(1)
```

Comment:
This part is for the actual fix. The previous code did not wait and immediately called the server again when the scoring server returned a non-success response, which caused the test failure.